File: src/main/java/dev/qixils/quasicolon/error/UserError.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package dev.qixils.quasicolon.error;

import dev.qixils.quasicolon.Key;
import dev.qixils.quasicolon.text.LocalizableText;
import org.checkerframework.checker.nullness.qual.NonNull;

public class UserError extends LocalizedRuntimeException {

	public UserError(@NonNull LocalizableText text) {
		super(text);
	}

	public UserError(@NonNull Key key, Object @NonNull ... args) {
		super(key, args);
	}
}

File: src/main/java/dev/qixils/quasicord/error/UserError.kt
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */
package dev.qixils.quasicord.error

import dev.qixils.quasicord.Key
import dev.qixils.quasicord.text.LocalizableText

open class UserError : LocalizedRuntimeException {
    constructor(text: LocalizableText) : super(text)

    constructor(key: Key, vararg args: Any) : super(key, *args)
}
